controllers/networktopology_test.go

```diff
@@ -1829,4 +1829,26 @@ var _ = Describe("NewNetworkTopologyReconciler", func() {
 			Expect(reconcileErr).NotTo(HaveOccurred())
 		})
 	})
+
+	When("the cluster is deleted and doesn't have our network topology finalizer", func() {
+		BeforeEach(func() {
+			actualCluster := &capi.Cluster{}
+			err := k8sClient.Get(ctx, request.NamespacedName, actualCluster)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(clusterClient.RemoveFinalizer(ctx, actualCluster, controllers.FinalizerNetTop)).To(Succeed())
+			Expect(actualCluster.Finalizers).NotTo(ContainElement(controllers.FinalizerNetTop))
+
+			Expect(clusterClient.AddFinalizer(ctx, actualCluster, "testing")).To(Succeed())
```

Comment:
I wasn't sure why we needed the `testing` finalizer at first, so maybe we can be more explicit about it.
```suggestion
			Expect(clusterClient.AddFinalizer(ctx, actualCluster, "keep-cr-so-we-can-test-deletion")).To(Succeed())
```